nexus/db-queries/src/db/datastore/affinity.rs

```diff
@@ -368,28 +402,78 @@ impl DataStore {
         &self,
         opctx: &OpContext,
         authz_anti_affinity_group: &authz::AntiAffinityGroup,
-        pagparams: &PaginatedBy<'_>,
-    ) -> ListResultVec<AntiAffinityGroupInstanceMembership> {
+        pagparams: &DataPageParams<'_, Uuid>,
+    ) -> ListResultVec<external::AntiAffinityGroupMember> {
         opctx.authorize(authz::Action::Read, authz_anti_affinity_group).await?;
 
-        use db::schema::anti_affinity_group_instance_membership::dsl;
-        match pagparams {
-            PaginatedBy::Id(pagparams) => paginated(
-                dsl::anti_affinity_group_instance_membership,
-                dsl::instance_id,
-                &pagparams,
-            ),
-            PaginatedBy::Name(_) => {
-                return Err(Error::invalid_request(
-                    "Cannot paginate group members by name",
-                ));
-            }
+        let asc = match pagparams.direction {
+            dropshot::PaginationOrder::Ascending => true,
+            dropshot::PaginationOrder::Descending => false,
+        };
+
+        let mut query = QueryBuilder::new()
+            .sql(
+                "SELECT id,label FROM (
+                SELECT instance_id as id, 'instance' as label
+                FROM anti_affinity_group_instance_membership
+                WHERE group_id = ",
+            )
+            .param()
+            .bind::<diesel::sql_types::Uuid, _>(authz_anti_affinity_group.id())
+            .sql(
+                "
+                UNION
+                SELECT affinity_group_id as id, 'affinity_group' as label
+                FROM anti_affinity_group_affinity_membership
+                WHERE anti_affinity_group_id = ",
+            )
```

Comment:
this use of `union` plus a label to zip together two tables is quite cute, it kinda makes me want to rewrite some code in my webhooks branch...